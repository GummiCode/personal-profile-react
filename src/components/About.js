import React from "react";
import styled from "styled-components";

import FullWidthContainer from './FullWidthContainer';
import HalfWidthContainer from './HalfWidthContainer';
import SectionArrow from './SectionArrow';

const AboutContainer = styled(FullWidthContainer)`
  background-color: #E5B168;
`;

const LeftAboutContainer = styled(HalfWidthContainer)`
  width: calc(50% - 40px);
  padding: 0 0 0 40px;
`

const RightSideScreen = styled(HalfWidthContainer)`
  width: 40vw;
  background: rgba(10, 10, 10, 0.3);
  clip-path: polygon(10% 0, 100% 0, 100% 100%, 0 100%);
`

const TitleText = styled.h1`
  font-size: 5em;
  color: #FFD9B4;
  padding: 0;
  margin: 30px 0 0 0;
`;

const SectionText = styled.ul`
  font-size: 2em;
  list-style: none;
`;

const Item = styled.li`
  margin: 0 0 4vh 0;
`;

const Bold = styled.span`
  font-weight: bold;
`;



const About  = ({ nextSection }) => {
  return (
    <AboutContainer
      data-testid="about"
      id="about"
    >
        <LeftAboutContainer>
          <TitleText>About:</TitleText>
          <SectionText>
            <Item>
              I'm a <Bold>Manchester-based junior developer</Bold>.
            </Item>
            <Item> 
              I'm seeking my first <Bold>Full-Time Position</Bold> as a <Bold>Software Engineer</Bold>.
            </Item>
            <Item>
              I've worked with <Bold>React</Bold>, <Bold>Express</Bold>, <Bold>MySQL</Bold>, <Bold>Heroku</Bold>, <Bold>React Testing Library</Bold> and other [contemporary] technologies.
            </Item>
            <Item>
              I've undertaken <Bold>Intensive Self-Training</Bold> using <Bold>Contemporary Resources</Bold> to ensure [relevance] of skills.
            </Item>
            <Item>
              I come from a career in <Bold>Chemical Research and Development</Bold>. I possess strong executive skills including <Bold>Project Management</Bold>, <Bold>Troubleshooting</Bold>, <Bold>Independent Learning</Bold> and <Bold>Collaborative Working</Bold>.
            </Item>
          </SectionText>
        </LeftAboutContainer>
        <RightSideScreen />

        <SectionArrow 
          nextSectionId={nextSection}
        />

    </AboutContainer>
  )
};

export default About;