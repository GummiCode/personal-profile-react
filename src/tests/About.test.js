import React from "react";
import { render } from "@testing-library/react";
import About from "../components/About";

test("renders the About component", () => {
  const { getByTestId } = render(<About />);
  const testId = getByTestId("about");
  expect(testId).toBeInTheDocument();
});

test("About component screenshot test", () => {
  const { asFragment } = render(<About />);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});
