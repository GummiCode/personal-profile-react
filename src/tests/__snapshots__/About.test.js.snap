// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component screenshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm sc-AxhCb letZQW"
    data-testid="about"
    id="about"
  >
    <div
      class="sc-AxirZ sc-AxhUy glbHQz"
    >
      <h1
        class="sc-AxheI drkDFn"
      >
        About:
      </h1>
      <ul
        class="sc-Axmtr fvVXkl"
      >
        <li
          class="sc-AxmLO fyzVOQ"
        >
          I'm a 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Manchester-based junior developer
          </span>
          .
        </li>
        <li
          class="sc-AxmLO fyzVOQ"
        >
          I'm seeking my first 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Full-Time Position
          </span>
           as a 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Software Engineer
          </span>
          .
        </li>
        <li
          class="sc-AxmLO fyzVOQ"
        >
          I've worked with 
          <span
            class="sc-fzozJi eWsAKO"
          >
            React
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Express
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            MySQL
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Heroku
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            React Testing Library
          </span>
           and other [contemporary] technologies.
        </li>
        <li
          class="sc-AxmLO fyzVOQ"
        >
          I've undertaken 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Intensive Self-Training
          </span>
           using 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Contemporary Resources
          </span>
           to ensure [relevance] of skills.
        </li>
        <li
          class="sc-AxmLO fyzVOQ"
        >
          I come from a career in 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Chemical Research and Development
          </span>
          . I possess strong executive skills including 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Project Management
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Troubleshooting
          </span>
          , 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Independent Learning
          </span>
           and 
          <span
            class="sc-fzozJi eWsAKO"
          >
            Collaborative Working
          </span>
          .
        </li>
      </ul>
    </div>
    <div
      class="sc-AxirZ sc-AxgMl jsusbt"
    />
    <a
      class="sc-AxiKw ixvQq"
    >
      <svg
        fill="currentColor"
        height="5em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(0, 0, 0);"
        viewBox="0 0 320 512"
        width="5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
